packages/rlc-common/src/buildSamples.ts

```diff
@@ -0,0 +1,835 @@
+  function dfs(root?: Schema) {
+    if (!root) {
+      return;
+    }
+    const parents = getImmediateParentsNames(root, [SchemaContext.Input]);
+    for (let parent of parents) {
+      if (isVisited.has(parent)) {
+        continue;
+      }
+      dfs(schemaMap.get(parent));
+      isVisited.add(parent);
+    }
+  }
+}
+
+function extractObjectProperties(
+  properties: Record<string, Schema>,
+  schemaMap: Map<string, Schema>,
+  allPropNames: Set<string>,
+  values: string[],
+  path: Set<string>
+) {
+  for (const prop in properties) {
+    const propName = prop;
+    const propertySchema = properties[prop];
+    if (propertySchema.readOnly) {
+      continue;
+    }
+    if (propertySchema.type === "never") {
+      continue;
+    }
+    if (allPropNames.has(propName)) {
+      continue;
+    }
+    const propType = propertySchema.typeName ?? propertySchema.type;
+    if (propType !== "string" && !schemaMap.has(propType)) {
+      schemaMap.set(propType, propertySchema);
+    }
+    allPropNames.add(propName);
+    let propRetValue =
+      `${propName}: ` +
+      mockParameterTypeValue(propType, propName, schemaMap, path);
+    values.push(propRetValue);
+  }
+}
+
+function isQuotedString(type: string) {
+  return (
+    /^"([a-zA-Z0-9=;\/\+\-\s]*)"$/g.test(type) ||
+    /^'([a-zA-Z0-9=;\/\+\-\s]*)'$/g.test(type)
+  );
+}
+
+function isRecord(type: string) {
+  return /^Record<([a-zA-Z].+),(\s*)([a-zA-Z].+)>$/g.test(type);
+}
+
+function getRecordType(type: string) {
+  return /Record<([a-zA-Z].+),(\s*)(?<type>[a-zA-Z].+)>/g.exec(type)?.groups
+    ?.type;
+}
+
+function isArray(type: string) {
+  return isArrayObject(type) || isNativeArray(type);
+}
+
+function isArrayObject(type: string) {
+  return /(^Array<([a-zA-Z].+)>$)/g.test(type);
+}
+
+function getArrayObjectType(type: string) {
+  return /Array<(?<type>[a-zA-Z].+)>/g.exec(type)?.groups?.type;
+}
+
+function isNativeArray(type: string) {
+  return /(^[a-zA-Z].+)\[\]$/g.test(type);
+}
+
+function getNativeArrayType(type: string) {
+  return /(?<type>[a-zA-Z].+)\[\]/g.exec(type)?.groups?.type;
+}
+
+function isUnion(type: string) {
+  const members = type.split("|").map((m) => m.trim());
+  return members.length > 1;
+}
+
+function getUnionType(type: string) {
+  return leaveBracket(type.split("|").map((m) => m.trim())[0]);
+}
+
+function leaveBracket(type: string) {
+  if (!type || type.length < 2) {
+    return type;
+  } else if (type.startsWith("(") && type.endsWith(")")) {
+    return type.slice(1, type.length - 1);
+  }
+  return type;
+}
+
+function leaveStringQuotes(str: string) {
+  const doubleQuotes = /^"(?<str>[a-zA-Z0-9=;\/\+\-\s]*)"$/g;
+  const singleQuote = /^'(?<str>[a-zA-Z0-9=;\/\+\-\s]*)'$/g;
+
+  return (
+    doubleQuotes.exec(str)?.groups?.str ??
+    singleQuote.exec(str)?.groups?.str ??
+    str
+  );
+}
+
+export enum TypeScriptType {
+  string,
+  date,
+  number,
+  boolean,
+  constant,
+  record,
+  array,
+  object,
+  union,
+  unknown,
+  enum
```

Comment:
I don't think we have a pure Uint8Array case for now, because for bytes content in body we will intercept as `string | Uint8Array | ReadableStream<Uint8Array> | NodeJS.ReadableStream`, then sample gen would guess this is a union type and would take the first type `string` to mock data.